web/checks_api.go

```diff
@@ -125,6 +123,61 @@ func ApiCreateChecksCatalogHandler(s services.ChecksService) gin.HandlerFunc {
 	}
 }
 
+// ApiCheckResultsHandler godoc
+// @Summary Get a specific cluster's check results
+// @Produce json
+// @Param cluster_id path string true "Cluster Id"
+// @Success 200 {object} map[string]interface{}
+// @Failure 500 {object} map[string]string
+// @Router /clusters/{cluster_id}/results [get]
+func ApiClusterCheckResultsHandler(s services.ChecksService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		clusterId := c.Param("cluster_id")
+
+		checkResults, err := s.GetChecksResultAndMetadataByCluster(clusterId)
+		if err != nil {
+			c.Error(err)
+			return
+		}
+
+		c.JSON(http.StatusOK, checkResults)
+	}
+}
+
+// ApiCreateChecksResultHandler godoc
+// @Summary Create a checks result entry
+// @Produce json
+// @Param id path string true "Resource Id"
+// @Param Body body JSONChecksResult true "Checks result"
+// @Success 201 {object} JSONChecksResult
+// @Failure 500 {object} map[string]string
+// @Router /checks/{id}/results [post]
```

Comment:
I have some doubts on the enpoint `/checks/{id}/results`

I see a different variant that makes explicit the Cluster concept.
`/clusters/:id/checks/results` or
`/clusters/:id/checks-results` or anything along these lines.

This is mainly because as of now Checks don't exist on their own, they're always somehow bound to a cluster (are they? :sweat_smile: )

Yet, no strict need to change anything if you wish, it is sort of a reminder to make discussions about API design.

Reply:
I knew this discussion would arise XD

The thing is that from the runner point of view, it doesn't really know that it is running cluster checks.
It only knows that there are some nodes and groups of nodes (that in our case, by now, are only clustered nodes) that have some assigned checks. With this data we create the inventory. And it will publish the results of this group.

If in the future we add other logical groups (like SAP system checks, individual nodes, SAP landscapes), this implementation would still have sense. If we use any kind of `clusters` endpoint, we would need to start diverging how we publish and store the results.

That's why I tried to make it generic.
In any case, it is true that by now we only have cluster checks, and at this moment naming the endpoint like `clusters/:cluster_id/results` would make sense.